myhoard/restore_coordinator.py

```diff
@@ -1128,13 +1137,38 @@ def _rename_prefetched_binlogs_back(self, binlogs):
                 last_renamed_index = remote_index - 1
         self.update_state(last_flushed_index=last_renamed_index, last_renamed_index=last_renamed_index)
 
+    def _sql_thread_not_running(self, slave_status):
+        return slave_status["Slave_SQL_Running"] != "Yes"
+
+    def _start_sql_thread(self, cursor):
+        # Try to start SQL thread with an incremental delay
+        if self.sql_thread_restart_count > 0:
+            factor = 1 if self.sql_thread_restart_count == 1 else (self.sql_thread_restart_count - 1) * 1.5
+            iteration_delay = min(self.SQL_THREAD_RESTART_INIT_INTERVAL * factor, self.SQL_THREAD_RESTART_MAX_INTERVAL)
+            if self.sql_thread_restart_time is not None:
+                time_passed = time.monotonic() - self.sql_thread_restart_time
+                if time_passed < iteration_delay:
+                    return
+        self.sql_thread_restart_time = time.monotonic()
+        cursor.execute("START SLAVE SQL_THREAD")
+
     def _check_sql_slave_status(self):
         expected_range = self.state["current_executed_gtid_target"]
         expected_index = self.state["current_relay_log_target"]
 
         with self._mysql_cursor() as cursor:
             cursor.execute("SHOW SLAVE STATUS")
             slave_status = cursor.fetchone()
+
+            # Sometimes mysqld can be OOM-killed during the backup restoration.
+            # In that case we should try restarting it.
+            if self._sql_thread_not_running(slave_status):
+                self._start_sql_thread(cursor)
+                cursor.execute("SHOW SLAVE STATUS")
```

Comment:
I'd make `_start_sql_thread` return boolean indicating whether it started the thread or not to avoid making this call when the information is not expected to be any different.

Reply:
Good point! done.